### PR TITLE
Feature/testcases/filter/apiintegration

### PR DESCRIPTION
**Pull request by AsmitaMargaje**
filter api integration test draft details file

### CODE REVIEW

File: src/screens/TestCases/Modal/CustomFilterModal.js [hunk 1]
```diff
@@ -110,44 +217,90 @@ const CustomFilterModal = ({
...
                           aria-label="Default select example"
+                          onChange={(e) => {
+                            setFilterType(e.target.value);
+                          }}
                         >
                           <option defaultValue>Open this select menu</option>
-                          <option value="1">Equals</option>
-                          <option value="2">Does Not Equals</option>
-                          <option value="3">Begins With</option>
+                          {filterColumn === 'id' ? (
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/Modal/CustomFilterModal.js:225]: all option store into a variable, then map it

File: src/screens/TestCases/Modal/CustomFilterModal.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import React, { useState, useRef, useEffect } from 'react';
 import { Overlay, Tooltip, Modal, Button } from 'react-bootstrap';
 import './CustomFilterStyle.css'; // Ensure this CSS file includes the .custom-container class
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/Modal/CustomFilterModal.js:3]: import in correct sequence as per doc

File: src/screens/TestCases/Modal/CustomFilterModal.js [hunk 3]
```diff
@@ -9,17 +11,26 @@ const CustomFilterModal = ({
...
+  setFilterText,
+  filterType,
+  columnName
 }) => {
   const [showTooltip, setShowTooltip] = useState(false);
   const [showModal, setShowModal] = useState(false);
   const target = useRef(null);
   const searchRef = useRef(null);
   const containerRef = useRef(null);
+  // const [filterType, setFilterType] = useState('');
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/Modal/CustomFilterModal.js:32]: remove all commented code, if not req. else give proper comment

File: src/screens/TestCases/Modal/CustomFilterModal.js [hunk 4]
```diff
@@ -45,6 +56,46 @@ const CustomFilterModal = ({
 
   const handleShow = () => setShowModal(true);
   const handleCloseModal = () => setShowModal(false);
+  const dispatch = useDispatch();
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/Modal/CustomFilterModal.js:59]: remove all commented code, if not req. else give proper comment

File: src/screens/TestCases/Modal/CustomFilterModal.js [hunk 5]
```diff
@@ -55,7 +106,7 @@ const CustomFilterModal = ({
       style={{ top: position.top, left: position.left }}
     >
       <div>
-        <b className="fs-6 text-custom">Submodule</b>
+        <b className="fs-6 text-custom">{columnName}</b>
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/Modal/CustomFilterModal.js:109]: what is test-custom class, give proper name

File: src/screens/TestCases/Modal/CustomFilterModal.js [hunk 6]
```diff
@@ -70,7 +121,7 @@ const CustomFilterModal = ({
 
         <hr className="my-1" />
         <p className="fs-6 mb-0 ms-4 cursor-pointer text-filters-container">
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/Modal/CustomFilterModal.js:123]: cursor-pointer class is not exist , why use it

File: src/screens/TestCases/Modal/CustomFilterModal.js [hunk 7]
```diff
@@ -88,9 +139,63 @@ const CustomFilterModal = ({
           }) => (
             <div className="custom-toolkit" {...props}>
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/Modal/CustomFilterModal.js:140]: why spread props  here

File: src/screens/TestCases/Modal/CustomFilterModal.js [hunk 8]
```diff
@@ -88,9 +139,63 @@ const CustomFilterModal = ({
...
               <div className="my-3">
-                <p className="mb-2 pointer" onClick={handleShow}>
-                  Equals
-                </p>
+                {filterColumn === 'id' ? (
+                  <>
+                    <p className="mb-2 pointer" onClick={handleShow}>
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/Modal/CustomFilterModal.js:144]: what is pointer class

File: src/screens/TestCases/Modal/CustomFilterModal.js [hunk 9]
```diff
@@ -110,44 +217,90 @@ const CustomFilterModal = ({
...
                         <input
-                          type="text"
+                          type={filterColumn === 'id' ? 'number' : 'text'}
                           className="form-control"
                           placeholder="Type"
+                          onChange={(e) => setFilterText(e.target.value)}
                         />
                       </div>
+                      {/* )} */}
+                      {/* {filterType === 'is between' && (
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/Modal/CustomFilterModal.js:268]: remove all commented code, if not req. else give a proper comment

File: src/screens/TestCases/TestDraft/TestDraftDetails.js [hunk 10]
```diff
@@ -79,106 +78,39 @@ function TestDraftDetails(props) {
...
+  const [filterColumnId, setFilterColumnId] = useState(null);
+
   const [modalPosition, setModalPosition] = useState({ top: 0, left: 0 });
   const [selectedFilters, setSelectedFilters] = useState([]);
-  const [filteredData, setFilteredData] = useState(
-    Array?.isArray(getDraftTestListData) ? getDraftTestListData : []
-  );
+
   const [searchTerm, setSearchTerm] = useState('');
-  // const closeModal = () => {
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/TestDraft/TestDraftDetails.js:88]: remove all commented code, if not req. else give a proper comment

File: src/screens/TestCases/TestDraft/TestDraftDetails.js [hunk 11]
```diff
@@ -79,106 +78,39 @@ function TestDraftDetails(props) {
...
+    type_name: 'type_id',
+    id: 'id',
+    severity: 'severity',
+    group_name: 'group_id',
+    steps: 'steps',
+    expected_result: 'expected_result',
+    status: 'status',
+    project_name: 'project_id'
+
+    // Add more mappings if needed
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/TestDraft/TestDraftDetails.js:102]: remove GPT comment

File: src/screens/TestCases/TestDraft/TestDraftDetails.js [hunk 12]
```diff
@@ -190,30 +122,55 @@ function TestDraftDetails(props) {
...
-  //   setFilteredData(newData);
-  //   closeModal();
-  // };
-
-  const handleApplyFilter = () => {
-    let newData = getDraftTestListData;
-    if (selectedFilters.length > 0 && filterColumn) {
-      newData = getDraftTestListData.filter((row) =>
-        selectedFilters.includes(row[filterColumn])
+  const [filterType, setFilterType] = useState('');
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/TestDraft/TestDraftDetails.js:125]: make it one state convert into use reducer hooks

File: src/screens/TestCases/TestDraft/TestDraftDetails.js [hunk 13]
```diff
@@ -932,35 +1084,23 @@ function TestDraftDetails(props) {
...
-          handleCheckboxChange={handleCheckboxChange}
-          selectedFilters={selectedFilters}
-          handleSelectAll={handleSelectAll}
-          uniqueValues={filteredUniqueValues}
-          searchTerm={searchTerm}
-          setSearchTerm={setSearchTerm}
-        />
-      )} */}
-
       {modalIsOpen && (
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/TestDraft/TestDraftDetails.js:1087]: why use

File: src/redux/slices/testCases/downloadFormatFile/index.js [hunk 14]
```diff
@@ -201,7 +201,14 @@ const downloadFormatSlice = createSlice({
       .addCase(getDraftTestCaseList.fulfilled, (state, action) => {
         state.isLoading.getDraftTestListData = false;
         state.successMsg.getDraftTestListData = action?.payload;
-        state.getDraftTestListData = action?.payload?.data?.data;
+        let data = !action?.payload?.data?.data
+          ? action?.payload?.data
+          : action?.payload?.data?.data;
+        // console.log('action', action?.payload?.data);
+        console.log('action1', action?.payload?.data?.data);
```
Comment (somyaranjan-techneai) [inline src/redux/slices/testCases/downloadFormatFile/index.js:208]: remove console

File: src/screens/TestCases/Modal/CustomFilterModal.js (hunk 1 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/Modal/CustomFilterModal.js]: resolved

File: src/screens/TestCases/Modal/CustomFilterModal.js (hunk 2 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/Modal/CustomFilterModal.js]: resolved

File: src/screens/TestCases/Modal/CustomFilterModal.js (hunk 3 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/Modal/CustomFilterModal.js]: resolved

File: src/screens/TestCases/Modal/CustomFilterModal.js (hunk 5 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/Modal/CustomFilterModal.js]: resolved

File: src/screens/TestCases/Modal/CustomFilterModal.js (hunk 6 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/Modal/CustomFilterModal.js]: resolved

File: src/screens/TestCases/Modal/CustomFilterModal.jsx [hunk 15]
```diff
@@ -0,0 +1,420 @@
...
+            className="check-box-size"
+            checked={selectedFilters.includes(value.label)} // Check if value.label is in selectedFilters
+            onChange={(e) => handleCheckboxChange(e, value.label, value.value)} // Pass label and value to handleFilterCheckboxChange
+          />
+          <label className="mx-3 fs-6 mb-0" htmlFor={`filter${value.value}`}>
+            {value.label} {/* Display the label property of value */}
+          </label>
+        </div>
+      ))}
+      <div className="button-container">
```
Comment (AmitSolnke) [inline src/screens/TestCases/Modal/CustomFilterModal.jsx:405]: Kindly make the size of btn small

File: src/screens/TestCases/Modal/CustomFilterModal.jsx
Comment (AmitSolnke) [inline src/screens/TestCases/Modal/CustomFilterModal.jsx:1]: 1.when column data displayed ,there should be overflow ,currently if list is long , the modal height is also increasing.

File: src/screens/TestCases/Modal/CustomFilterModal.jsx [hunk 16]
```diff
@@ -0,0 +1,420 @@
...
+
+  // <option value="ends with">End with</option>
+  // <option value="does not end with">Does Not End with</option>
+
+  // <option value="contains"> Contains</option>
+  // <option value="does not contain">
+  //   Does Not Contain
+  // </option>
+
+  useEffect(() => {
```
Comment (AmitSolnke) [inline src/screens/TestCases/Modal/CustomFilterModal.jsx:107]: Mention Purpose of this code

File: src/screens/TestCases/Modal/CustomFilterModal.jsx [hunk 17]
```diff
@@ -0,0 +1,420 @@
...
+      style={{ top: position.top, left: position.left }}
+    >
+      <div>
+        <b className="fs-6 text-custom">{columnName}</b>
+        <hr className="my-2" />
+
+        <p className="Sort mb-2">Sort</p>
+        <span className="d-flex flex-column">
+          <p
+            className="fs-6 mb-2 pointer"
```
Comment (AmitSolnke) [inline src/screens/TestCases/Modal/CustomFilterModal.jsx:187]: why 2 class with same purpose? 
1.Pointer 
2.Cusrsor-Pointer

User "cp" which globally avaible class for same purpose

File: src/screens/TestCases/Modal/CustomFilterModal.js (hunk 7 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/Modal/CustomFilterModal.js]: its help to open the text filter and number filter modal

File: src/screens/TestCases/Modal/CustomFilterModal.js (hunk 8 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/Modal/CustomFilterModal.js]: resolved

File: src/screens/TestCases/Modal/CustomFilterModal.js (hunk 9 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/Modal/CustomFilterModal.js]: resolved

File: src/screens/TestCases/TestDraft/TestDraftDetails.js (hunk 11 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/TestDraft/TestDraftDetails.js]: resolved

File: src/screens/TestCases/TestDraft/TestDraftDetails.js (hunk 10 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/TestDraft/TestDraftDetails.js:88]: resolved

File: src/screens/TestCases/TestDraft/TestDraftDetails.js (hunk 12 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/TestDraft/TestDraftDetails.js]: resolved

File: src/redux/slices/testCases/downloadFormatFile/index.js (hunk 14 above)
Comment (AsmitaMargaje) [reply src/redux/slices/testCases/downloadFormatFile/index.js]: resolved

File: src/screens/TestCases/Modal/CustomFilterModal.jsx (hunk 17 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/Modal/CustomFilterModal.jsx:189]: resolved

File: src/screens/TestCases/Modal/CustomFilterModal.jsx (hunk 16 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/Modal/CustomFilterModal.jsx:109]: to show the filter modal inside the filter icon

File: src/screens/TestCases/Modal/CustomFilterModal.jsx (hunk 15 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/Modal/CustomFilterModal.jsx:405]: resolved

File: src/screens/TestCases/Modal/CustomFilterModal.jsx
Comment (AsmitaMargaje) [reply src/screens/TestCases/Modal/CustomFilterModal.jsx:1]: resolved

File: src/screens/TestCases/TestDraft/TestDraftDetails.js (hunk 13 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/TestDraft/TestDraftDetails.js:1228]: resolved

File: src/redux/services/testCases/downloadFormatFile/index.js [hunk 18]
```diff
@@ -66,33 +62,33 @@ export const getSubModuleMasterThunk = createAsyncThunk(
...
+          .map((id) => `submodule_id[]=${id}`)
+          .join('&');
+        endpoint += `&${submoduleQueryParam}`;
+      }
 
-      const endpoint = `draftFile/getTestdraftBulkFormat?project_id=${project_id}&module_id=${module_id}&submodule_id=${encodeURIComponent(
-        submoduleParam
-      )}`;
       const response = await customAxios.get(endpoint);
+      console.log('rrr', response);
```
Comment (somyaranjan-techneai) [inline src/redux/services/testCases/downloadFormatFile/index.js:79]: Remove console

File: src/redux/services/testCases/downloadFormatFile/index.js [hunk 19]
```diff
@@ -66,33 +62,33 @@ export const getSubModuleMasterThunk = createAsyncThunk(
...
-        if (response?.data?.status === 1) {
-          onSuccessHandler();
-          toast.success(response?.data?.message);
-          return response?.data?.message;
-        } else {
-          // onErrorHandler();
-          errorHandler(response);
-        }
+        toast.success('File Downloaded Successfully');
+        // if (response?.data?.status === 1) {
```
Comment (somyaranjan-techneai) [inline src/redux/services/testCases/downloadFormatFile/index.js:83]: Remove commented  code, if not needed

File: src/redux/slices/testCases/downloadFormatFile/index.js [hunk 20]
```diff
@@ -201,7 +213,13 @@ const downloadFormatSlice = createSlice({
       .addCase(getDraftTestCaseList.fulfilled, (state, action) => {
         state.isLoading.getDraftTestListData = false;
         state.successMsg.getDraftTestListData = action?.payload;
-        state.getDraftTestListData = action?.payload?.data?.data;
+        let data = !action?.payload?.data?.data?.data
```
Comment (somyaranjan-techneai) [inline src/redux/slices/testCases/downloadFormatFile/index.js:216]: use nullish coalescing

File: src/redux/slices/testCases/testCaseReview/index.js [hunk 21]
```diff
@@ -41,7 +49,8 @@ const testCaseReviewSlice = createSlice({
       })
       .addCase(getTestCaseReviewListThunk.fulfilled, (state, action) => {
         state.isLoading.testCaseReviewList = false;
-        state.testCaseReviewList = action?.payload?.data;
+        state.testCaseReviewList = action?.payload?.data?.data?.data;
+        state.filterTestCaseReviewList = action?.payload?.data?.filter_data;
         state.successMsg.testCaseReviewList = action?.payload?.message;
```
Comment (somyaranjan-techneai) [inline src/redux/slices/testCases/testCaseReview/index.js:54]: are you sure the message key coming from payload

File: src/screens/TestCases/FunctionMaster/FunctionMasterComponent.jsx [hunk 22]
```diff
@@ -77,7 +75,7 @@ function FunctionMasterComponent() {
           )}
           {row.is_active == 0 && (
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/FunctionMaster/FunctionMasterComponent.jsx:76]: use ternary operator

File: src/screens/TestCases/Modal/CustomFilterModal.jsx [hunk 23]
```diff
@@ -0,0 +1,348 @@
...
+                            placeholder="Type"
+                            onChange={(e) =>
+                              handleBetweenValueChange(0, e.target.value)
+                            }
+                          />
+                        </div>
+                      )}
+                      {(filterType === 'is between' ||
+                        filterType === 'is not between') && (
+                        <>
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/Modal/CustomFilterModal.jsx:256]: why use pragment

File: src/screens/TestCases/Modal/CustomFilterModal.jsx [hunk 24]
```diff
@@ -0,0 +1,348 @@
...
+                    </Button>
+                  </Modal.Footer>
+                </Modal>
+              </div>
+            </div>
+          )}
+        </Overlay>
+        <hr className="my-1" />
+
+        <div>
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/Modal/CustomFilterModal.jsx:286]: what is this

File: src/screens/TestCases/Modal/CustomFilterModal.jsx [hunk 25]
```diff
@@ -0,0 +1,348 @@
...
+          <div className="position-relative">
+            <input
+              ref={searchRef}
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/Modal/CustomFilterModal.jsx:289]: why use ref

File: src/screens/TestCases/TestingGroupMaster/TestingGroupMasterComponent.jsx [hunk 26]
```diff
@@ -73,7 +79,7 @@ function TestingGroupMasterComponent() {
           )}
           {row.is_active == 0 && (
             <span className="badge bg-danger" style={{ width: '4rem' }}>
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/TestingGroupMaster/TestingGroupMasterComponent.jsx:81]: why use inline style

File: src/screens/TestCases/TestingTypeMaster/AddTestingTypeModal.jsx [hunk 27]
```diff
@@ -1,43 +1,43 @@
...
+    if (type === 'ADD') {
       dispatch(
         addTestingTypeMasterThunk({
           formData: formData,
           onSuccessHandler: () => {
             close();
             dispatch(getTestingTypeMasterListThunk());
           },
-          onErrorHandler: () => {},
+          onErrorHandler: () => {}
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/TestingTypeMaster/AddTestingTypeModal.jsx:40]: why use

File: src/screens/TestCases/TestingTypeMaster/AddTestingTypeModal.jsx [hunk 28]
```diff
@@ -49,7 +49,7 @@ function AddTestingTypeModal({ show, close, type, currentTestingTypeData }) {
...
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/TestingTypeMaster/AddTestingTypeModal.jsx:52]: why use

**Pull request merged**